src/runtimes/node/dynamic_imports/parser.js

```diff
@@ -0,0 +1,140 @@
+
+const acorn = require('acorn')
+
+const ECMA_VERSION = 2017
+const GLOB_WILDCARD = '*'
+
+// Transforms an array of glob nodes into a glob string including an absolute
+// path.
+//
+// Example: ["./files/", "*", ".json"] => "/home/ntl/files/*.json"
+const getAbsoluteGlob = ({ basePath, globNodes, resolveDir }) => {
+  if (!validateGlobNodes(globNodes)) {
+    return null
+  }
+
+  const globStarIndex = globNodes.indexOf(GLOB_WILDCARD)
+  const [staticPath, dynamicPath] =
+    globStarIndex === -1
+      ? [globNodes.join(''), '']
+      : [globNodes.slice(0, globStarIndex).join(''), globNodes.slice(globStarIndex).join('')]
+  const absoluteStaticPath = resolve(resolveDir, staticPath)
+  const relativeStaticPath = relative(basePath, absoluteStaticPath)
+  const absoluteGlob = join(relativeStaticPath, dynamicPath)
+
+  return absoluteGlob
+}
+
+// Tries to parse an expression, returning an object with:
+// - `includedPathsGlob`: A glob with the files to be included in the bundle
+// - `type`: The expression type (e.g. "require", "import")
+const parseExpression = ({ basePath, expression, resolveDir }) => {
+  const { body } = acorn.parse(expression, { ecmaVersion: ECMA_VERSION })
+  const { callee } = body[0].expression
+
+  if (callee.name === 'require') {
+    const includedPathsGlob = parseRequire({ basePath, expression: body[0].expression, resolveDir })
+
+    if (includedPathsGlob) {
+      return {
+        includedPathsGlob,
+        type: callee.name,
+      }
+    }
+  }
+}
+
+// Parses a `require()` and returns a glob string with an absolute path.
+const parseRequire = ({ basePath, expression, resolveDir }) => {
+  const { arguments: args } = expression
+  const argType = args[0].type
+
+  if (argType === 'BinaryExpression') {
+    try {
+      const globNodes = parseBinaryExpression(args[0])
+
+      return getAbsoluteGlob({ basePath, globNodes, resolveDir })
+    } catch (_) {
+      // no-op
+    }
+  }
+
+  if (argType === 'TemplateLiteral') {
+    const globNodes = parseTemplateLiteral(args[0])
+
+    return getAbsoluteGlob({ basePath, globNodes, resolveDir })
+  }
+}
+
+// Transforms a binary expression AST node into an array of glob nodes, where
+// static parts will be left untouched and identifiers will be replaced by
+// `GLOB_WILDCARD`.
+//
+// Example: './files/' + lang + '.json' => ["./files/", "*", ".json"]
+const parseBinaryExpression = (expression) => {
+  const { left, operator, right } = expression
+
+  if (operator !== '+') {
+    throw new Error('Expression operator not supported')
+  }
+
+  const operands = [left, right].map((operand) => {
+    switch (operand.type) {
+      case 'BinaryExpression':
+        return parseBinaryExpression(operand)
+
+      case 'CallExpression':
+      case 'Identifier':
+        return GLOB_WILDCARD
+
+      case 'Literal':
+        return operand.value
+
+      default:
+        throw new Error('Expression member not supported')
+    }
+  })
+
+  return operands.flat()
```

Comment:
[sand] `flatMap()` could be used instead of `map()` + `flat()`

Reply:
Good catch! Done in https://github.com/netlify/zip-it-and-ship-it/pull/521/commits/eab6fdb2ef7cf70883ba9eec81cd7f2cf473450b. Thanks. 🙌🏻